keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/api/ec2/ClusterDependencyContainer.kt

```diff
@@ -0,0 +1,74 @@
+package com.netflix.spinnaker.keel.api.ec2
+
+import com.netflix.spinnaker.keel.api.Locatable
+import com.netflix.spinnaker.keel.api.ResourceSpec
+import com.netflix.spinnaker.keel.api.SubnetAwareLocations
+import com.netflix.spinnaker.keel.core.api.ClusterDependencies
+
+/**
+ * A [ResourceSpec] that has default and override cluster dependencies (i.e. security groups, load
+ * balancers, and/or target groups).
+ */
+interface OverrideableClusterDependencyContainer<T : ClusterDependencyContainer> :
+  Locatable<SubnetAwareLocations> {
+  val defaults: T
+  val overrides: Map<String, T>
```

Comment:
I recommend adding a comment to indicate that the key is a region name (if that is, indeed, the case).